object-store-client-play-26/src/main/scala/uk/gov/hmrc/objectstore/client/play/PlayWSHttpClient.scala

```diff
@@ -90,12 +90,22 @@ class PlayWSHttpClient @Inject()(wsClient: WSClient)(implicit ec: ExecutionConte
       .withQueryStringParameters(queryParameters: _*)
       .withRequestTimeout(Duration.Inf)
 
-      body
-        .writeBody(wsRequest)
-        .stream()
-        .map(logResponse)
-        .map(processResponse)
-        .andThen { case _ => body.release() }
+      val res =
+        body
+          .writeBody(wsRequest)
+          .stream()
+          .map(logResponse)
+          .map(Right(_): Either[PlayObjectStoreException, WSResponse])
+          .recover {
+            case e: TimeoutException => Left(GatewayTimeoutException(e))
```

Comment:
Can we not simply return a failed Future here wrapping our exceptions and not using Either?

Reply:
Or you might just be able to use `F.raiseError(e)` here and remove the Either

Reply:
It would have been nice to implement this in `F` directly, using `F.raiseError` but this seemed clearer since the bulk of it is in Future.
I didn't want to embed the errors into the future, and then dig them out again.